src/Plugin/AbstractFX.hpp

```diff
@@ -65,6 +74,38 @@ class AbstractPluginFX : public Plugin
         delete effect;
         delete filterpar;
     }
+    
+    virtual void doReinit(const bool firstInit)
+    {
+        // save current param values before recreating effect
+        uchar params[paramCount];
+
+        if (effect != nullptr)
+        {
+            for (int i=0, count=static_cast<int>(paramCount); i<count; ++i)
+                params[i] = effect->getpar(i+2);
+
+            delete effect;
+        }
+
+        zyn::EffectParams pars(allocator, false, efxoutl, efxoutr, 0, static_cast<uint>(sampleRate), static_cast<int>(bufferSize), filterpar);
+        
+        effect = instantiateFX(pars);
+
+        if (firstInit)
+        {
+            effect->setpreset(0);
+        }
+        else
+        {
+            for (int i=0, count=static_cast<int>(paramCount); i<count; ++i)
+                effect->changepar(i+2, params[i]);
```

Comment:
This is old code, but I wonder: is `params` used uninitialized here if effect was `nullptr` above?

Reply:
@fundamental do you have an idea here?

Reply:
Hm, a second reading through it looks like effect _should_ only be a nullptr if and only if it's the firstInit. Maybe as a guard make firstInit non-const and put an else statement at line 87 which forces it to true if the effect was null?

Reply:
done